tests/test_fixed_stress.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import porepy as pp
+import pytest
+from porepy.applications.test_utils.models import add_mixin
+
+import pp_solvers
+from pp_solvers.dof_manager import DofManager
+from pp_solvers.preconditioners import SinglePhysicsPreconditioner
+
+from pp_solvers.fixed_stress import (
+    make_fs_analytical_slow_new,
+    get_fixed_stress_stabilization,
+    get_fs_fractures_analytical,
+)
+
+
+@pytest.fixture(scope="module", params=[False, True])
+def with_fractures(request) -> bool:
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def model(with_fractures) -> pp.PorePyModel:
+    """Instantiate a model for the test suites in this file."""
+
+    class TailoredClass(
+        pp_solvers.IterativeSolverMixin,
+        pp.model_geometries.SquareDomainOrthogonalFractures,
+        pp.Thermoporomechanics,
+    ):
+        """Common base class for all models in this test suite."""
+
+        def meshing_arguments(self):
+            return {"cell_size": self.params["cell_size"]}
+
+    params = {
+        "linear_solver": {},
+        "cell_size": 0.25,
+        "cartesian": True,
+        "fracture_indices": [0, 1] if with_fractures else [],
+        # A non-zero fluid comressibility is needed for the fracture fixed stress.
+        "material_constants": {
+            "fluid": pp.FluidComponent(
+                compressibility=1,
+            ),
+        },
+    }
+    model = TailoredClass(params=params)
+    model.prepare_simulation()
+    model.before_nonlinear_loop()
+    model.before_nonlinear_iteration()
+    model.assemble_linear_system()
+    return model
+
+
+def test_fixed_stress(model: pp_solvers.IterativeSolverMixin, with_fractures: bool):
+    jacobian = model.bmat
+
+    # The fixed stress in fractures requires a non-zero u_intf jump.
+    interfaces = model.mdg.interfaces(dim=model.nd - 1)
+    u_intf = model.interface_displacement(interfaces)
+    u_intf_values = model.equation_system.get_variable_values([u_intf], iterate_index=0)
+    u_intf_values[:] = np.arange(u_intf_values.size)
+    model.equation_system.set_variable_values(
+        values=u_intf_values, variables=[u_intf], iterate_index=0
+    )
+
+    # YZ: Is there a better way than hard-coding these numbers?
```

Comment:
In principle, there should be if we get a better understanding of how we really think about the ordering of blocks relative to a nested preconditioner. Should we sit down and discuss at some point?

Reply:
Definitely